packages/use-contractkit/src/utils/useProviders.ts

```diff
@@ -0,0 +1,97 @@
+import { useMemo } from 'react';
+import { isMobile } from 'react-device-detect';
+
+import {
+  localStorageKeys,
+  Priorities,
+  PROVIDERS,
+  WalletTypes,
+} from '../constants';
+import { Maybe, Provider, WalletConnectProvider, WalletEntry } from '../types';
+import localStorage from './localStorage';
+import { defaultProviderSort } from './sort';
+
+export function walletToProvider(wallet: WalletEntry): WalletConnectProvider {
+  return {
+    name: wallet.name,
+    type: WalletTypes.WalletConnect,
+    description: wallet.description || 'Missing description in registry',
+    icon: wallet.logos.md,
+    canConnect: () => true,
```

Comment:
Is this right? I'm seeing some providers where this isn't the case.

Reply:
I'm not sure I understand what are you asking is right. Could you clarify a bit?